parsl/monitoring/monitoring.py

```diff
@@ -245,7 +247,7 @@ def __del__(self):
             self.priority_msgs.put(("STOP", 0))
 
     def close(self):
-        return self.__del__()
+        self.__del__()
```

Comment:
I think the style for this should be `__del__` calls `close`, not `close` calls `__del__`

I'm still a little concerned about the ordering of events at process exit / garbage collection time: will we definitely get something back from the Hub? Or can it have been destroyed before __del__ is called? (i.e. what causes this __del__ to be called before whatever gets rid of the other end of the queue?)

Reply:
I switched the `__del__` and `close`. 

I think your concern is right. Currently it works okay so far. But we certainly need to make sure everything for monitoring is garbage collected later. I will create a new issue #963 to keep track of that.